impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java

```diff
@@ -319,6 +337,17 @@ else if (this.clientConfig.getClientCredentialsResolver() == null) {
             this.clientConfig.setBaseUrlResolver(new DefaultBaseUrlResolver(this.clientConfig.getBaseUrl()));
         }
 
+        // OAuth2
+        if (this.clientConfig.getAuthenticationScheme() == AuthenticationScheme.OAUTH2) {
```

Comment:
Why do you need both `AuthenticationScheme` and `AuthorizationMode` to determine if you have to retrieve an access token?

Reply:
AuthorizationMode == "PrivateKey" is sufficient? I can get rid of AuthenticationScheme in config?

Reply:
Will get rid of AuthenticationScheme config property.